src/meshes/p4est_mesh.jl

```diff
@@ -48,7 +55,14 @@ mutable struct P4estMesh{NDIMS, RealT <: Real, IsParallel, P, Ghost, NDIMSP2, NN
         ghost = ghost_new_p4est(p4est)
         ghost_pw = PointerWrapper(ghost)
 
-        mesh = new{NDIMS, eltype(tree_node_coordinates), typeof(is_parallel),
+        # To enable the treatment of a manifold of dimension NDIM embedded within an
+        # ambient space of dimension NDIMS_AMBIENT, we store both as type parameters and
+        # allow them to differ in the general case. This functionality is used for
+        # constructing discretizations on spherical shell domains for applications in
+        # global atmospheric modelling. The ambient dimension NDIMS_AMBIENT is therefore 
+        # set here in the inner constructor to size(tree_node_coordinates, 1).
+        mesh = new{NDIMS, size(tree_node_coordinates, 1),
```

Comment:
I'm not completely happy with this since it's another type instability. However, we're already not type stable due to MPI (and it's at a high level and not the lower-level kernels), so this is probably fine for now.

Reply:
(To be clear, there is nothing to do here right now, it's just something I wanted to write down explicitly)

Reply:
Agreed, that makes sense to me.